packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -4415,11 +4415,14 @@ namespace Tpetra {
         auto lclRowMap = this->rowMap_->getLocalMap ();
         auto lclColMap = this->colMap_->getLocalMap ();
 
-        // KJ: We need this fence even though it passes tests. Map uses kokkos
-        // view with relying on UVM (it does not have a dual view context). So,
-        // the values can be touched somewhere and we need to make sure the
-        // updates should be done before we use it.
-        execution_space().fence ();
+        // FENCE REVIEW - NOW PASSING
+        //   Testing: This code is exercised by unit tests.
+        //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+        //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+        //   Plan:    Remove this fence.
+        //   Notes:   determineLocalTriangularStructure is device so we don't
+        //            need to fence here.
+        // execution_space().fence ();
```

Comment:
Regarding this fence:  I propose removing this entire branch of code and all the fields that it computes.  This branch computes constants lowerTriangular_ , upperTriangular_, and nodeNumDiags_.  I do not see anyplace in Trilinos where these are used after they are set. And I don't see an accessor that would let a user get these values.  Also, #2630 says they are inconsistent.
@trilinos/ifpack2 people:  do you need lowerTriangular_ , upperTriangular_  and nodeNumDiags_?

Reply:
@kddevin Maybe I'm missing it, but I don't see any accessor method for {lower|upper}Triangular_ or nodeNumDiags_.  As far as I can see, none of the three are used in Ifpack2.